.github/actions/setup/action.yml

```diff
@@ -0,0 +1,48 @@
+name: Setup
+description: Common setup steps used by our workflows
+runs:
+  using: composite
+  steps:
+    - name: Setup pnpm
+      uses: pnpm/action-setup@v2
+      with:
+        version: 8.6.5
+
+    - name: Setup node with nvmrc
+      uses: actions/setup-node@v3
+      with:
+        node-version-file: .nvmrc
+        registry-url: https://registry.npmjs.org
+        cache: pnpm
+
+    - name: Get foundry version
+      id: foundry-version
+      run: echo ::set-output name=version::$(cat .foundryrc)
+
+    - name: Setup foundry
+      uses: foundry-rs/foundry-toolchain@v1
+      with:
+        version: $${ steps.foundry-version.outputs.version }
+
+    - name: Install node dependencies
+      shell: bash
+      run: pnpm install --frozen-lockfile
+
+    # Workaround for parallel building with forge:
+    # we need to run a build on a single package to make sure
+    # that no race conditions happen with multiple forge instances
+    # later when we call nx build
+    # see https://github.com/foundry-rs/foundry/issues/4736
+    - name: Forge issue 4736 workaround
+      shell: bash
+      working-directory: packages/contracts-bedrock
+      run: forge build
+
+    - name: Derive appropriate SHAs for base and head for `nx affected` commands
+      uses: nrwl/nx-set-shas@v3
```

Comment:
This is how github actions derive the nx shas. This is necessary to always have nx commands reliably work